### PR TITLE
fix: fix Badge Wave borderColor error

### LINKED ISSUE #39178: Badge borderColor error
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/headless-pine-y0wsth?file=/src/App.tsx)

### Steps to reproduce

https://codesandbox.io/s/headless-pine-y0wsth?file=/src/App.tsx

### What is expected?

波纹伪元素的 border 颜色跟着主元素的背景颜色保持一致

### What is actually happening?

没有保持一致，after伪元素的颜色固定为 token.colorPrimary

image

| Environment | Info |
| --- | --- |
| antd | 5.0.3 |
| React | 18.2.0 |
| System | mac |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

- close #39178 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix: fix Badge borderColor error |
| 🇨🇳 Chinese | fix: 修复了自定义颜色的时候，波纹的颜色不会跟着小圆点颜色发生变化的问题 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by li-jia-nan** [comment]
### before:

image
image
image

### after:

image
image
image

File: components/badge/style/index.tsx [hunk 1]
```diff
@@ -193,7 +194,9 @@ const genSharedBadgeStyle: GenerateStyle<BadgeToken> = (token: BadgeToken): CSSO
             insetInlineStart: 0,
             width: '100%',
             height: '100%',
-            border: `${badgeShadowSize}px solid ${token.colorPrimary}`,
+            borderWidth: `${badgeShadowSize}px`,
```
Comment (zombieJ) [inline components/badge/style/index.tsx:197]: 为啥要拆开，而不是 ``` `${badgeShadowSize}px solid inherit` ```

File: components/badge/index.tsx [hunk 2]
```diff
@@ -153,7 +153,8 @@ const Badge: CompoundedComponent = ({
 
   const statusStyle: React.CSSProperties = {};
   if (color && !isPresetColor(color)) {
-    statusStyle.background = color;
+    statusStyle.color = color;
+    statusStyle.backgroundColor = color;
```
Comment (zombieJ) [inline components/badge/index.tsx:157]: 保持 `background`，还能省一些字~

**Review by zombieJ — APPROVED** [review]

File: components/badge/style/index.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/badge/style/index.tsx:197]: 复合属性写 inherit 不起作用，所以拆开了

File: components/badge/index.tsx (hunk 2 above)
Comment (li-jia-nan) [reply components/badge/index.tsx]: 改回去了

**Comment by afc163** [comment]
666

**Pull request merged**